src/odemis/acq/stream/_base.py

```diff
@@ -1302,7 +1302,8 @@ def getPixelCoordinates(self, p_pos):
         tform = AffineTransform(matrix, translation)
         pixel_pos_c = tform.inverse().apply(p_pos)
         # a "-" is used for the y coordinate because Y axis has the opposite direction in physical coordinates
-        pixel_pos = int(pixel_pos_c[0] + size[0] / 2), - int(pixel_pos_c[1] - size[1] / 2)
+        pixel_pos = (int(math.floor(pixel_pos_c[0] + size[0] / 2)),
+                     int(math.floor(- (pixel_pos_c[1] - size[1] / 2))))
         if 0 <= pixel_pos[0] < size[0] and 0 <= pixel_pos[1] < size[1]:
             return pixel_pos
```

Comment:
According to the docstring, the returned value could be a single `int`. I don't see how that can happen from the code (not related to your change by the way).

Reply:
The docstring is supposed to mean it'll return either a tuple[int, int] or a None. However, the "," makes it confusing. I've changed it to use the Python typing format, which is a lot less ambiguous.